#  An Experiment From Scratch: using QPDAC
## Question, Plan, Data, Analysis, Conclusion
To go through a complete pipeline of experiment design, we first raise a question; plan the metrics and conditions we would like to study; collect the data; conduct an analysis; come to a conclusion.  
In this notebook I will design an experiment from scratch -- from raising a random question: *Does Google show marketplace pages in both normal and cognito mode?*


The metric we use here is simply a binary variable indicating whether the result page from Google Search shows marketplace pages. We shall run an experiment to decide if two modes are the same.   
The components of the designed experiment are listed below:  

* Factors: viewer modes  
* Levels: standard, incognito
* Conditions: {normal mode, incognito mode}

The way we collect the data is intuitive:

* Google search 37 common outwears(listed in "common outwear" page in Wikipedia) and record if the result page has marketplaces for the searched item
* Perform t-test on the difference of probabilities of two modes seeing marketplaces in the result page

**37 Common Outwears used as search keyword**    
*Abaya, Academic gown, Anorak, Apron, Blazer, Cagoule, Cloak, Coat, Duffle coat, Duster, Frock coat, Gilet, Goggle jacket, Greatcoat, Hat, Hoodie, Jacket, Leather jacket, Matchcoat, Mess jacket, Opera coat, Overcoat, Pea coat, Poncho, Raincoat, Rain, pants, Redingote, Robe, Shawl, Shrug, Ski suit, Sleeved blanket, Sport coat, Top coat, Touque, Trench coat, Windbreaker*

State a hypothesis tests 

$H_0$ : $p_1$ - $p_2$ = 0, the probabilities of seeing marketplaces in the result page when searching common outwears are the same for normal and incognito mode.  
$H_1$ : $p_1$ - $p_2$ $\neq$ 0, the probabilities of seeing marketplaces in the result page when searching common outwears are not the same for normal and incognito mode.

Calculate the p-value and power of the test and make a decision

In [15]:
import pandas as pd
import scipy
import math
data = pd.read_csv('expdes_hw1_data.csv')

In [16]:
p_inco = data['incognito'].mean()
p_normal = data['normal'].mean()
data['diff'] = data['incognito'] - data['normal']
delta_mean = data['diff'].mean()
data['diff'].describe()

count    38.000000
mean     -0.368421
std       0.589138
min      -1.000000
25%      -1.000000
50%       0.000000
75%       0.000000
max       1.000000
Name: diff, dtype: float64

In [29]:
p_s = data['normal'].mean()
p_i = data['incognito'].mean()
p = (37 * p_s + 37 * p_i) / 74
z_score = (p_s - p_i) / math.sqrt((p_s * (1 - p_s) / 37) + ( p_i * (1 - p_i) / 37))
print(f'z-score = {z_score}')
p_value = scipy.stats.norm.sf(abs(z_score)) * 2
print(f'p value = {p_value}')

z-score = 3.9619401430321615
p value = 7.434318057386893e-05


In [33]:
CO = 1.96
delta = 0.368421
sigma = 0.589138
power = 2 * (1 - scipy.stats.norm(0, 1).cdf(CO - (delta / sigma)))
print(f'power = {power}') 

power = 0.18199289931673834


* P Value = 0.00007
* Power = 0.18  
* Given such significant P value and such low power, I would rather be more conservative and still conclude that there's no difference in search result pages from incognito mode or normal mode. Increasing number of observations might help.  